### ISSUE TITLE
not-an-iterable in openai chat completion

### ISSUE THREAD

**Issue by lutzroeder** [microsoft/pyright#6772]
**Describe the bug**

See openai/openai-python#870 or code below.

**Code or Screenshots**

1. `pip install openai`
2. Open code in `.py` file in Visual Studio Code
```
import openai

client = openai.OpenAI()

with open('test.txt', 'r', encoding='utf-8') as handle:
    contents = handle.read()

completion = client.chat.completions.create(
    model="gpt-4-1106-preview",
    messages = [ { "role":"user", "content": "Summarize\n\n" + contents } ],
    stream = True
)

for event in completion:
    content = event.choices[0].delta.content
    print(content, end='', flush=True)
```

<img width="433" alt="screenshot" src="https://github.com/microsoft/pyright/assets/438516/2f397f7b-3286-4685-a7d8-363ef5a8c358">

**Comment by erictraut** [comment]
I'm not able to repro the problem as reported. Which version of the `openai` library are you using? I installed the latest published version from pypi, which is 1.5.0. I also tried to repro the problem with 1.3.5 (which is the version used in the linked bug report), and I also didn't see any type checking errors.

Also, which version of pyright are you using?

Please confirm that you're able to repro the problem with the latest version of openai and pyright and the code sample above.

**Comment by hmc-cs-mdrissi** [comment]
The error message is not a pyright error. It's pylint one. You should report it to [pylint](https://github.com/pylint-dev/pylint/issues) if it reproduces on latest version.

**Comment by erictraut** [comment]
Ah, good catch @hmc-cs-mdrissi. I didn't look closely enough at the error message.

Closing since this isn't a pyright issue.
